helpers/Authoring.php

```diff
@@ -42,40 +42,29 @@ class Authoring
 {
 
     /**
-     * Validate and format (if possible) a QTI XML string
+     * Validate and format (if possible) a QTI XML string.
+     * 
+     * QTI XML string will be sanitized (if possible invalid elements will be removed).
      * 
      * @param string $qti
      * @return string
      * @throws QtiModelException
      */
     public static function validateQtiXml($qti){
+        
+        $sanitizedQti = self::sanitizeQtiXml($qti);
+        
+        $dom = self::loadQtiXml($sanitizedQti);
+        $returnValue = $dom->saveXML();
 
-        $returnValue = '';
-
-        // render and clean the xml
-        $dom = new DOMDocument('1.0', 'UTF-8');
-        $dom->formatOutput = true;
-        $dom->preserveWhiteSpace = false;
-        $dom->validateOnParse = false;
-
-        if($dom->loadXML($qti)){
-            $returnValue = $dom->saveXML();
-
-            //in debug mode, systematically check if the save QTI is standard compliant
-            if(DEBUG_MODE){
-                $parserValidator = new Parser($returnValue);
-                $parserValidator->validate();
-                if(!$parserValidator->isValid()){
-                    common_Logger::w('Invalid QTI output: '.PHP_EOL.' '.$parserValidator->displayErrors());
-                    common_Logger::d(print_r(explode(PHP_EOL, $returnValue), true));
-                    throw new QtiModelException('invalid QTI item XML '.PHP_EOL.' '.$parserValidator->displayErrors());
-                }
-            }
-        }else{
-            $parserValidator = new Parser($qti);
+        //in debug mode, systematically check if the save QTI is standard compliant
+        if(DEBUG_MODE){
```

Comment:
since the validation is already in loadQtiXml, why not simply try and catch the potential exception and throw a user friendly on an upper level ?